models/users.js

```diff
@@ -458,6 +459,10 @@ const getRdsUserInfoByGitHubUsername = async (githubUsername) => {
  */
 
 const getUsersBasedOnFilter = async (query) => {
+  if (query.state === "ONBOARDING" && query.time) {
+    const fetchUsersWithOnBoardingState = await getUsersWithOnboardingState(query);
+    return fetchUsersWithOnBoardingState;
+  }
```

Comment:
what if I want to get onboarding users along with other filters. 🤔 If you return from here other filters wont work

Reply:
tried fixing, have a look

Reply:
@heyrandhir we need to keep this condition `query.state === "ONBOARDING" && query.time` because for IDLE and ACTIVE states we cannot be checking from discordJoined date, Hence we need to check for this condition strictly.